src/components/DonateForm.jsx

```diff
@@ -0,0 +1,181 @@
+          this.cardElement.clear();
+          // Push donation to firebase
+          this.props.firebase
+            .push('donations', {
+              amount,
+              createdAt: response.data.created,
+              stripeId: response.data.id,
+              note,
+              uid,
+            });
+        } else if (response.status === 200) {
+          // Card declined
+          this.setState({ error: response.data.message });
+        } else {
+          // Stripe error
+          this.setState({ error: response });
+        }
+
+        this.setState({
+          isLoading: false,
+          isProcessing: false
+        });
+      })
+    }).catch((error) => {
+      // Axios error
+      this.setState({
+        error: 'Please try again later.',
+        isLoading: false,
+        isProcessing: false
+      });
+      console.error(error);
+    });
+  }
+
+  render() {
+    return (
+      <form className='grid-item item-s-12 item-l-9 item-xl-6 no-gutter' onSubmit={event => event.preventDefault()}>
+        <div className='grid-row'>
+          <div className='grid-item item-s-12 margin-bottom-small'>
+            <div className='border-top'></div>
+          </div>
+
+          <div className='grid-item item-s-12 grid-row align-items-center no-gutter margin-bottom-small'>
+            <div className='grid-item flex-grow'>
+              <input
+                id='amount'
+                name='amount'
+                type='range'
+                min='1'
+                max='100'
+                step='1'
+                disabled={this.state.isLoading}
+                value={this.state.amount}
+                onChange={ event => this.setState({ amount: event.target.value })}
+              />
+            </div>
+            <div className='grid-item item-s-3 item-m-2 font-size-large'>${this.state.amount} <span className='font-size-small'>USD</span></div>
+            <label htmlFor='amount' className='grid-item item-s-12 margin-top-micro font-size-small u-inline-block'>
+              <div>Give from $1 - $100</div>
+            </label>
+          </div>
+
+          <label className='grid-item item-s-12 margin-bottom-small'>
+            <CardElement
+              onReady={(ref) => this.cardElement = ref}
+              onChange={(element) => {
+                // Card element form is filled
+                this.setState({ complete: element.complete })
+              }}
+              style={{base: {fontSize: '14px'}}}
+            />
+            <div className='margin-top-micro font-size-small'>Card details</div>
+          </label>
+
+          <div id="form-odie-description" className='grid-item item-s-12 margin-bottom-small'>
+            <textarea
+              id='note'
+              name='note'
+              placeholder='note'
+              disabled={this.state.isLoading}
+              value={this.state.note}
+              onChange={ event => this.setState({ note: event.target.value })}
```

Comment:
this is unsanitized text which might be something we have to check?

Reply:
Yes we should

Reply:
https://stackoverflow.com/questions/6659351/removing-all-script-tags-from-html-with-js-regular-expression